frontend/src/components/AssignmentModal/ContactInfoCard.vue

```diff
@@ -0,0 +1,418 @@
+            + Add new contact
+          </v-btn>
+        </v-col>
+      </v-row>
+    </v-container>
+  </v-card-text>
+</template>
+
+<script setup lang="ts">
+  import { defineEmits, reactive, computed, onMounted } from 'vue';
+  import { storeToRefs } from 'pinia';
+
+  import { type IFormRules } from '@/utils/interfaces';
+  import { useAssignmentStore } from '@/stores/assignment';
+  import { useConfirmationStore } from '@/stores/confirmation';
+
+  import CustomInput from '@/components/UI/CustomInput.vue';
+  import CustomSelect from '@/components/UI/CustomSelect.vue';
+  import CustomAutocomplete from '@/components/UI/CustomAutocomplete.vue';
+
+  import { 
+    contactTypesLabels,
+    phoneNumberTypesLabels, 
+    addressesTypesLabels, 
+    statesCodesLabels 
+  } from '@/helpers/assignmentModal';
+
+  const { assignmentState } = storeToRefs(useAssignmentStore());
+  const { 
+    addContactData, 
+    addContactInfoData,
+    removeContactData,
+    removePhoneNumberData,
+    removeAddressData
+  } = useAssignmentStore();
+  const { setConfirmationDataAndShow } = useConfirmationStore();
+
+  const emits = defineEmits(['validate-form']);
+
+  onMounted(() => {
+    emits('validate-form');
+  })
+
+  const getContactTypesLabels = computed(
+    () => contactTypesLabels.filter(contactType => !assignmentState.value.contactsInfo.some(
+        contact => contact.type?.includes(contactType) && 
+        (contact.type.toLowerCase().includes('owner') ||
+        contact.type.toLowerCase().includes('driver') ||
+        contact.type.toLowerCase().includes('estimator'))
+      ))
+  );
+
+  const setAssignmentDataPropName = (infoType: string) => {
+    const infoTypeInLowerCase = infoType.toLowerCase();
+
+    if (infoTypeInLowerCase.includes('phone')) {
+      return 'phoneNumbers';
+    } else if (infoTypeInLowerCase.includes('address')) {
+      return 'addresses';
+    }
+
+    return '';
+  }
+
+  const addContact = () => {
+    const sortedContactsArr = 
+    assignmentState.value.contactsInfo.sort((a,b) => a.id - b.id);
+
+    const qtyOfContacts = sortedContactsArr.length;
+    const lastContactIdValue = sortedContactsArr[qtyOfContacts - 1].id;
+
+    addContactData({
+      id: lastContactIdValue + 1,
+      type: getContactTypesLabels.value[0],
+      firstName: '',
+      lastName: '',
+      email: '',
+      phoneNumbers: [{
+        id: 0,
+        type: 'Mobile',
+        number: ''
+      }],
+      addresses: [{
+        id: 0,
+        type: 'Home',
+        city: '',
+        state: null,
+        zip: '',
+        addressLine: ''
+      }]
+    });
+
+    emits('validate-form');
+  }
+
+  const addInfo = (contactId: number, infoType: string) => {
+    const contactIndex = assignmentState.value.contactsInfo.findIndex(contact => contact.id == contactId);
+    const assignmentDataPropName = setAssignmentDataPropName(infoType);
+
+    if (contactIndex > -1 && assignmentDataPropName) {
+      const sortedInfoArr = 

```

Comment:
sortedInfoArr is a bad variable name. It is unclear what data is inside of sortedInfoArr

Reply:
Removed it